ete-test/ete.py

```diff
@@ -0,0 +1,336 @@
+@pytest.fixture
+def remove_output_information(flagged):
+	return flagged.delete_schema()
+
+#***********************************************************************************************************#
+#OUTPUT DATA#####################################################################################OUTPUT DATA#
+#***********************************************************************************************************#
+
+#Returns processed data (flagged table)
+@pytest.fixture
+def flagged_data(flagged):
+	return flagged.get_full_table()
+
+@pytest.fixture
+def flags_data(flags):
+	return flags.get_full_table()
+
+#***********************************************************************************************************#
+#HELPER TESTS###################################################################################HELPER TESTS#
+#############################################################################################################
+
+#Tests valid client instance creation
+def test_client(client):
+	assert isinstance(client, _Client)
+
+#Test valid ctran instance creation
+def test_ctran(ctran):
+	assert isinstance(ctran, CTran_Data)
+
+#Test valid flagged instance creation
+def test_flagged(flagged):
+	assert isinstance(flagged, Flagged_Data)
+
+#Test valid flags instance creation
+def test_flags(flags):
+	assert isinstance(flags, Flags)
+
+#Tests saving test data to aperture
+def test_save_ctran_test_data(save_ctran_test_data):
+	assert save_ctran_test_data == True
+
+#Test hive creation, for output/processed data
+def test_create_hive(create_hive):
+	assert create_hive == True
+
+#Tests data processing and saving [function in client]
+def test_process_and_save(process_and_save):
+	assert process_and_save == True
+
+#Test that correct number of rows has been inserted
+def test_pull_test_data(get_ctran_test_data):
+	assert not type(get_ctran_test_data) is None
+	assert len(get_ctran_test_data) == 5
+
+#Tests pulling processed data for further testing
+def test_pull_flagged_data(flagged_data):
+	assert not type(flagged_data) is bool
+	assert len(flagged_data) > 0
+
+#Test pulling flags data
+def test_pull_flags(flags_data):
+	assert not type(flags_data) is bool
+	assert len(flags_data) > 0
+
+#***********************************************************************************************************#
+#ACTUAL TESTS###################################################################################ACTUAL TESTS#
+#############################################################################################################
+
+#Test first (input) row: contains all None's, except date.
+#Row_id can't be null because it's automatically incremented
+def test_row_1(flagged_data, flags_data):
+	#Step 1: Need to split flags to have only _NULL flags
+	mask = flags_data.description.str.contains("_NULL")
+	null_flags = flags_data[mask]
+
+	#Should have more than 1 NULL flag
+	assert len(null_flags) > 1
```

Comment:
Why is this > 1? Shouldnt we know the exact amount of nulls its expecting?

Reply:
If they add datafields in the future, and increment number of null flags, we won't know for sure number of nulls. I can change it though

Reply:
Please do, tests should be designed for what it is now. If they change it later and break a test it'll they're responsible to go and edit the test